packages/grid/_modules_/grid/hooks/features/rows/useGridRows.ts

```diff
@@ -83,14 +86,14 @@ export const useGridRows = (
   React.useEffect(() => {
     setGridState((state) => {
       internalRowsState.current = convertGridRowsPropToState(
-        rows,
-        state.options.rowCount,
-        getRowId,
+        props.rows,
+        props.rowCount,
+        props.getRowId,
       );
 
       return { ...state, rows: internalRowsState.current };
     });
-  }, [getRowId, rows, setGridState]);
+  }, [props.getRowId, props.rows, props.rowCount, setGridState]);
```

Comment:
I think you can have it as just `props` here since you are depending on all 3.

```suggestion
  }, [props, setGridState]);
```